src/UserStory/Entries.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react'
+import { Link } from 'react-router-dom'
+import validate from 'validate.js'
+
+const initialState = {
+    storyFields: { date:"", owner:"", usId:"", title:"", question:"", note:"" },
+    fieldErrors: { dateError:"", ownerError:"", usIdError:"", titleError:"" }
+}
+
+const constraints = {
+    date: {
+        presence: {
+            allowEmpty : false
+        }
+    },
+    owner: {
+        presence: {
+            allowEmpty : false
+        }
+    },
+    usId: {
+        presence: {
+            allowEmpty : false
+        }
+    },
+    title: {
+        presence: {
+            allowEmpty : false
+        }
+    }
+}
+
+class Entries extends Component {
+    constructor(props){
+        super(props)
+        this.state = initialState;
+    }
+
+    validate() {
+        const error = validate(this.state.storyFields, constraints)
+
+        if(error){ 
+            this.setState({
+                dateError: error.date,
+                ownerError: error.owner,
+                usIdError: error.usId,
+                titleError: error.title
+            })
```

Comment:
In my opinion, you don't need to keep state for each field error. Just set the state for the whole error object. Ex: `this.setState(error)`

Reply:
noted, changes made

Reply:
```suggestion
validate() {
        const error = validate(this.state.storyFields, constraints)
        if(error){ 
            this.setState({fieldErrors: error});
            console.log(this.state.fieldErrors.dateError)
            return false
        }
        return true
    }
```
returns undefined

```suggestion
validate() {
        const error = validate(this.state.storyFields, constraints)
        if(error){ 
            this.setState({
                dateError: error.date,
                ownerError: error.owner,
                usIdError: error.usId,
                titleError: error.title
            })
            console.log(this.state.dateError)
            return false
        }
        return true
    }
```
returns "date cant be blank"

Im not sure if the errors are getting assigned to the fieldErrors, or im console.logging the wrong thing. There is no error output displayed when form is empty. Can you advise?